lib/src/configuration.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019 Google LLC. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:source_span/source_span.dart';
+
+import 'ast/node.dart';
+import 'ast/sass.dart';
+import 'util/limited_map_view.dart';
+import 'util/unprefixed_map_view.dart';
+import 'value.dart';
+
+/// A set of variables meant to configure a module by overriding its
+/// `!default` declarations.
+///
+/// There are two types of configuration: an explicit one that's created by
+/// the `with` clause of a `@use` rule and an implicit one that's created any
+/// time a file containing a `@forward` rule is imported.
+///
+/// Both types of configuration pass through `@forward` rules, but explicit
+/// configurations will cause an error if attempting to use them on a module
+/// that has already been loaded, while implicit configurations will be silently
+/// ignored in this case.
+class Configuration {
+  /// A map from variable names (without `$`) to values.
+  ///
+  /// When this is empty, it may be unmodifiable, so [Map.remove] should not be
+  /// called on this.
+  final Map<String, ConfiguredValue> values;
+
+  /// Whether or not this configuration is implicit.
+  final bool isImplicit;
```

Comment:
I might move the discussion of what an implicit configuration means from the class documentation to this getter.